src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs

```diff
@@ -12,6 +11,329 @@ namespace System.Text.RegularExpressions
     /// <summary>Detects various forms of prefixes in the regular expression that can help FindFirstChars optimize its search.</summary>
     internal static class RegexPrefixAnalyzer
     {
+        /// <summary>Cache of ToString() strings for the ASCII chars.</summary>
+        /// <remarks>The strings are lazily created on first use.</remarks>
+        private static string[]? s_asciiCharStrings;
+
+        /// <summary>Gets the ToString() string for the specified char.</summary>
+        private static string GetCharString(char ch)
```

Comment:
I of course trust your judgement, but would it really be that impactful if we decide to just precompute this array at construction time given we are only really covering ASCII? Or is it that we don't think that this array will get used most of the time?